go.mod

```diff
@@ -1,12 +1,12 @@
 module github.com/ethereum-optimism/optimism
 
-go 1.22.0
+go 1.23.0
 
-toolchain go1.22.7
+toolchain go1.24.7
```

Comment:
I don't think we should deviate the `toolchain`'s version from the `go` version.  So if we update to `go 1.23.0`, then we should probably have a toolchain in the `go1.23.` family.

Reply:
Oh god I don't know how this happened. Going to try to roll this back, I have successfully avoided doing this up until the rebase I did, somehow it got messed up during that.